### PR TITLE
Filtering for RBE actions

### PR DESCRIPTION
**Pull request by pariparajuli**
Filter/search with actionDigest for RBE actions

### CODE REVIEW

**Comment by pariparajuli** [comment]
Screen Shot 2021-08-05 at 4 38 33 PM
Screen Shot 2021-08-05 at 4 39 52 PM
Screen Shot 2021-08-05 at 4 42 02 PM

File: app/invocation/invocation_execution_card.tsx [hunk 1]
```diff
@@ -260,6 +261,15 @@ export default class ExecutionCardComponent extends React.Component {
         incompleteCount++;
       }
     }
+    console.log(this.props.filter);
```
Comment (bduffany) [inline app/invocation/invocation_execution_card.tsx:264]: nit: remove `console.log`

File: app/invocation/invocation_execution_card.tsx [hunk 2]
```diff
@@ -260,6 +261,15 @@ export default class ExecutionCardComponent extends React.Component {
...
+    const filteredActions = this.state.executions.filter(
+      (action) =>
+        !this.props.filter ||
+        `${action.actionDigest.hash}/${action.actionDigest.sizeBytes}`
```
Comment (bduffany) [inline app/invocation/invocation_execution_card.tsx:268]: Seems useful to be able to filter by the command, e.g. `go`, `bash`, `tsc`, ... if it doesn't slow down the search by too much, mind adding that here too?

Comment (pariparajuli) [reply app/invocation/invocation_execution_card.tsx:268]: I added that and made the placeholder for the filter input a bit different so it's clear what the user can filter by

File: app/invocation/invocation_execution_card.tsx [hunk 3]
```diff
@@ -316,42 +326,48 @@ export default class ExecutionCardComponent extends React.Component {
...
                     </div>
-                  </div>
-                </div>
-              );
-            })}
+                  );
+                })}
+              </div>
+            ) : (
+              <div>No matching actions.</div>
```
Comment (bduffany) [inline app/invocation/invocation_execution_card.tsx:369]: Sorry I didn't notice this on the last round -- I think maybe we should show "No actions found" if the user is not filtering?

**Review by bduffany — APPROVED** [review]

File: app/invocation/invocation_execution_card.tsx (hunk 3 above)
Comment (bduffany) [reply app/invocation/invocation_execution_card.tsx:369]: Resolved offline -- when no actions are found there's a separate message that gets displayed.

**Pull request merged**